core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -912,22 +947,43 @@ private void updateSchema(final GraphConfig config) {
                         .build();
                 addSchema(newSchema);
             }
+
+            if (null != config.getLibrary() && config.getLibrary().exists(config.getGraphId())) {
+                // Set Schema if null.
+                final Pair<Schema, StoreProperties> pair = config.getLibrary().get(config.getGraphId());
+                schema = (null == schema) ? pair.getFirst() : schema;
+            }
         }
 
-        private void updateStoreProperties(final GraphConfig config) {
-            StoreProperties mergedStoreProperties = null;
+        private void updateStoreProperties(final GraphConfig config, final Store store) {
+            final StoreProperties mergedStoreProperties = new StoreProperties();
+
+            //Add Properties from ParentProperties
             if (null != parentStorePropertiesId) {
-                mergedStoreProperties = config.getLibrary().getProperties(parentStorePropertiesId);
+                mergedStoreProperties.merge(config.getLibrary().getProperties(parentStorePropertiesId));
             }
 
+            //Add Properties from Builder
             if (null != properties) {
-                if (null == mergedStoreProperties) {
-                    mergedStoreProperties = properties;
-                } else {
-                    mergedStoreProperties.merge(properties);
+                mergedStoreProperties.merge(properties);
+            }
+
+            //Add Properties from Store
+            if (null != store) {
+                final StoreProperties storeProperties = store.getProperties();
+                if (null != storeProperties && mergedStoreProperties.getProperties().isEmpty()) {
+                    mergedStoreProperties.merge(storeProperties);
+                }
+            }
+
+            // Add Properties from Library
+            final boolean isLibraryPresent = null != config.getLibrary();
+            if (mergedStoreProperties.getProperties().isEmpty() && isLibraryPresent) {
+                final boolean hasLibraryPropertiesForGraphId = config.getLibrary().exists(config.getGraphId());
+                if (hasLibraryPropertiesForGraphId) {
+                    mergedStoreProperties.merge(config.getLibrary().get(config.getGraphId()).getSecond());
                 }
             }
-            properties = mergedStoreProperties;
```

Comment:
As a result of this change, the entire method does nothing at all and has no effect on anything - `mergedStoreProperties` used only inside this method and is never assigned to anything outside.
The previous version of this method only had an effect if `parentStorePropertiesId` was set. But there are no tests to ensure the parent properties are actually merged in, so removing this functionality doesn't cause any test failures.

This PR adds a new code for getting properties from the store and moves the code for getting properties from the library into here. But I think if changes are made to this then new tests are needed to ensure everything works as intended.

Reply:
hilarious! yes that was meant to be last line in the method.

Reply:
yep, no one tested as thoroughly as the FederatedStore :D